### PR TITLE
[bug] Avoid deleting endpoint port when running on Replit

### PR DESCRIPTION
**Pull request by blast-hardcheese**
- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired changed?
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Does your submission pass the tests? 
- [x] Have you successfully ran tests with your changes locally?

This adds an additional gate to @cdmistman's original PR, to ensure that we do not thwart reverse-proxy setups.

Note, the original suggestion was to modify the backend, similar to how codespaces works, but I didn't realize I was attempting to test with @cdmistman's change directly applied to `.web/`. When I attempted to verify the fix in a fresh environment, the original issue popped up again, cementing in my mind that this is the correct fix.

Second attempt at #3441

### CODE REVIEW

**Comment by blast-hardcheese** [comment]
There are two unrelated changes to the python backend, even though this should be a three-line change:

1. When tracing through the codespaces path to figure out where I needed to implement the agreed-upon change, it became clear that we had some confused booleans that needed some help.
2. When rolling through `CONTRIBUTING.md`, `black` suggested some formatting changes. Happy to revert those if you'd prefer.

File: reflex/.templates/web/utils/state.js [hunk 1]
```diff
@@ -101,7 +101,11 @@ export const getBackendURL = (url_str) => {
       } else if (endpoint.protocol === "http:") {
         endpoint.protocol = "https:";
       }
-      endpoint.port = ""; // Assume websocket is on https port via load balancer.
+      // If we are running on replit.dev, avoid deleting the port.
```
Comment (cdmistman) [inline reflex/.templates/web/utils/state.js:104]: nice. for context, in "development mode" repls are available at `<repl id>-<random string>.replit.dev`

**Review by masenf — COMMENTED** [review]
i thought there was an environment var that we could read on the backend to set the `api_url` appropriately. did that not end up working?

**Comment by masenf** [comment]
i just tried this in `config.py` and it seems to work for me without the `state.js` changes

```python
        if "api_url" not in self._non_default_attributes:
            # If running in Github Codespaces or repl.it, override API_URL
            codespace_name = os.getenv("CODESPACE_NAME")
            GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN = os.getenv(
                "GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN"
            )
            replit_dev_domain = os.getenv("REPLIT_DEV_DOMAIN")
            backend_port = kwargs.get('backend_port', self.backend_port)
            if codespace_name and GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN:
                self.api_url = (
                    f"https://{codespace_name}-{backend_port}"
                    f".{GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}"
                )   
            elif replit_dev_domain:
                self.api_url = (
                    f"https://{replit_dev_domain}:{backend_port}"
                )
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -80,10 +80,10 @@ poetry run pyright reflex tests
...
 
-Finally, run `black` to format your code.
+Finally, run `ruff` to format your code.
 
 ``` bash
-poetry run black reflex tests
+poetry run ruff format .
 ```
 
 Consider installing git pre-commit hooks so Ruff, Pyright, Darglint and Black will run automatically before each commit.
```
Comment (Lendemor) [inline CONTRIBUTING.md:89]: Thanks for catching the inconsistency in `CONTRIBUTING.md`.

```suggestion
Consider installing git pre-commit hooks so Ruff, Pyright, Darglint and `make_pyi` will run automatically before each commit.
```

**Review by Lendemor — CHANGES_REQUESTED** [review]

**Comment by blast-hardcheese** [comment]
> i just tried this in `config.py` and it seems to work for me without the `state.js` changes

I can only expect some of my problems diagnosing this were due to using Brave as my primary browser, even with "Shields" down it seems to have been having difficulty navigating establishing the websocket.

A fresh environment, a fresh repl, using Chrome, and I have confirmed that what we originally discussed does in fact work. I have reflected that in this PR, please pardon the back and forth.

**Review by masenf — APPROVED** [review]
Excellent 😀 This will be in our next release 0.5.4 🎉

**Comment by masenf** [comment]
@Lendemor needs your re-approval to merge

**Review by Lendemor — APPROVED** [review]

**Pull request merged**